SDL/MiniDoublet.cuh

```diff
@@ -92,25 +90,503 @@ namespace SDL
+                drdz = modulesInGPU.drdzs[moduleIndex];
+                // printf("drdz: %f\n", drdz);
+            }
+            else
+            {
+                // printf("moduleIndex: %d\n", moduleIndex);
+                // printf("partnerModuleIndex: %d\n", modulesInGPU.partnerModuleIndex(moduleIndex));
+                drdz = modulesInGPU.drdzs[modulesInGPU.partnerModuleIndices[moduleIndex]];
+                // printf("drdz: %f\n", drdz);
+            }
+        }
+        else
+        {
+            drdz = 0;
+        }
+        const float miniTilt = ((isTilted) ? 0.5f * pixelPSZpitch * drdz / alpaka::math::sqrt(acc, 1.f + drdz * drdz) / moduleGapSize(modulesInGPU,moduleIndex) : 0);
+
+        // Compute luminous region requirement for endcap
+        const float miniLum = alpaka::math::abs(acc, dPhi * deltaZLum/dz); // Balaji's new error
+        // const float miniLum = abs(deltaZLum / lowerHit.z()); // Old error
+
+
+        // =================================================================
+        // Return the threshold value
+        // =================================================================
+        // Following condition is met if the module is central and flatly lying
+        if (modulesInGPU.subdets[moduleIndex] == Barrel and modulesInGPU.sides[moduleIndex] == Center)
+        {
+            return miniSlope + alpaka::math::sqrt(acc, miniMuls * miniMuls + miniPVoff * miniPVoff);
+        }
+        // Following condition is met if the module is central and tilted
+        else if (modulesInGPU.subdets[moduleIndex] == Barrel and modulesInGPU.sides[moduleIndex] != Center) //all types of tilted modules
+        {
+            return miniSlope + alpaka::math::sqrt(acc, miniMuls * miniMuls + miniPVoff * miniPVoff + miniTilt * miniTilt * miniSlope * miniSlope);
+        }
+        // If not barrel, it is Endcap
+        else
+        {
+            return miniSlope + alpaka::math::sqrt(acc, miniMuls * miniMuls + miniPVoff * miniPVoff + miniLum * miniLum);
+        }
+    }
 
     ALPAKA_FN_ACC void shiftStripHits(struct modules& modulesInGPU, uint16_t& lowerModuleIndex, uint16_t& upperModuleIndex, unsigned int lowerHitIndex, unsigned int upperHitIndex, float* shiftedCoords,float xLower,float yLower,float zLower,float rtLower,float xUpper,float yUpper,float zUpper,float rtUpper);
 
-    ALPAKA_FN_ACC bool runMiniDoubletDefaultAlgo(struct modules& modulesInGPU, uint16_t& lowerModuleIndex, uint16_t& upperModuleIndex, unsigned int lowerHitIndex, unsigned int upperHitIndex, float& dz, float& dphi, float& dphichange, float& shiftedX, float& shiftedY, float& shiftedZ, float& noShiftedDz, float& noShiftedDphi, float& noShiftedDphiChange,float xLower,float yLower,float zLower,float rtLower,float xUpper,float yUpper,float zUpper,float rtUpper);
+    template<typename TAcc>
+    ALPAKA_FN_HOST_ACC ALPAKA_FN_INLINE float ATan2_alpaka(TAcc const & acc, float y, float x)
```

Comment:
This and some other functions originally come from `Hit.cuh`. Adding them here is a temporary measure to make things work and then they will be moved to their proper file? Can't we add the alpaka versions to the original file of the cuda variants?

Reply:
Yes, these are temporary. That sounds reasonable, I will move them over.